src/popup/components/ConsensusReachedLabel.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <span class="consensu-reached-label">
+    <template v-if="isConsensusReached">
+      {{ $t('multisig.approved') }}
+    </template>
+    <template v-else>
+      <PendingIcon class="icon" />
+      {{ $t('multisig.consensusPending') }}
+    </template>
+    <ConsensusLabel
+      :confirmations-required="confirmationsRequired"
+      :has-pending-transaction="hasPendingTransaction"
+      :default-confirmed-by="defaultConfirmedBy"
+      :confirmed-by="confirmedBy"
+      :signers="signers"
+      :has-dot="isConsensusReached"
+    />
+    <template v-if="isConsensusReached">
+      {{ $t('multisig.waitingToBeSent') }}
+    </template>
+  </span>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType } from '@vue/composition-api';
+import ConsensusLabel from './ConsensusLabel.vue';
+import PendingIcon from '../../icons/animated-pending.svg?vue-component';
+
+export default defineComponent({
+  components: {
+    ConsensusLabel,
+    PendingIcon,
+  },
+  props: {
+    signers: { type: Array as PropType<string[]>, required: true },
+    confirmedBy: { type: Array as PropType<string[]>, default: () => [] },
+    confirmationsRequired: { type: Number, required: true },
+    defaultConfirmedBy: { type: Number, default: 0 },
+    hasPendingTransaction: Boolean,
+  },
+  setup(props) {
+    const localConfirmedBy = computed(
+      () => (props.hasPendingTransaction)
+        ? props.confirmedBy?.length
+        : props.defaultConfirmedBy,
+    );
+
+    const isConsensusReached = computed(
```

Comment:
The component name is `Consesus Reached...` so by reading it I would assume that I should display it when the consensus is ready. But the logic encapsulated here states something opposite. So in my opinion the general naming convention is misleading here.